### ISSUE TITLE
Retry button for Paypal payments

### ISSUE THREAD

**Issue by alexanmtz** [worknenjoy/gitpay#256]
When a paypal payment is failed or open due to no concluded transaction, we should have a button to allow users to retry payment.

### Why
Users with the payment failed should have a chance to retry that same payment rather than create another one

### Level
Beginner

### Current behavior
When a payment is failed, we store the checkout url for the user and we should be able to allow the user to try again

### Expected behavior
- When the paypal payment failed, on the orders tab we should have a button retry with a reload icon which will redirect the user to checkout.

### usefull links
  - If you see this task: https://gitpay.me/#/task/39 you will see in orders tab a lot of Paypal payments open, so we should have a retry button in this case (in failed cases as well) to retry. Don't show any button in successful cases.

### Screenshot attached
![orders-gitpay](https://user-images.githubusercontent.com/88840/50699815-389d4a00-1049-11e9-96ca-af7e1986537e.png)

### How to solve this issue
 - Got to the task component
 - Add to orders tabs for Paypal payments a retry button
 - Link the retry button to the user checkout URL

### How to reproduce the user scenario
- Create a user
- Create a task
- Add a bounty with Paypal
- Cancel the checkout
- You will see a pending payment which should have the retry button

### What you will need to complete this issue
You will need the development environment keys to test the Paypal payments.

### Tips
We need a button on failed and pending payments that directs the user to the Paypal checkout url as we do for the first payment. This information is stored in the order.

**Comment by ajmorozoff** [comment]
Hey, I'd like to take this.

**Comment by alexanmtz** [comment]
@ajmorozoff please confirm your interest https://gitpay.me/#/task/213/ in "I'm interested" button, so I can assign you and you can receive a bounty once your changes is merged.

And you can join the Gitpay team here on Github: https://bit.ly/2Irhfqk 
And the slack channel if you need any help: https://bit.ly/31fXhr7

**Comment by VictorAlessander** [comment]
I found a possible bug in file **paypal-payment-dialog.js** on line 18.
The conditional is checking the value of state property **termsPaypal**, but doesn't exists.

I'm right about my analysis? If yes, I fix this in the same issue or I need to create another?
![Screenshot from 2019-07-07 13-03-19](https://user-images.githubusercontent.com/21283066/60771011-e8ff1000-a0b7-11e9-8eaf-ac640ff117c6.png)

**Comment by alexanmtz** [comment]
@VictorAlessander , great catch about the bug, actually this was probably after a huge refactoring we did. Thanks for looking into this one! You can send the PR together as you did, and let's review that 👍

**Comment by alexanmtz** [comment]
Hey @VictorAlessander , we merged your solution and it was deployed 🚀 

Check it here: https://gitpay.me/#/task/245, you can see that I've created a Paypal payment that was cancelled, and it works expected, but there's a big issue:

The only user that should be able to see the button is the logged user who created the order. So can you please send a new Pull Request allowing only the logged user to pay? You can do this by checking if the user is logged and if the user id from that order is the same that the logged user.

**Comment by VictorAlessander** [comment]
Of course.
I will do this right now.

**Comment by alexanmtz** [comment]
merged, deployed and now work as expected  🎉and the bounty will be send 💸@VictorAlessander , thanks for the great contribution 🙏
